sw/snRuntime/src/start_banshee.c

```diff
@@ -31,6 +31,14 @@ void _snrt_init_team(uint32_t cluster_core_id, uint32_t cluster_core_num,
     team->cluster_mem.start = spm_start;
     team->cluster_mem.end = spm_end;
 
+    // Allocate and initialize global barrier
+    // TODO: Atomics to global memory are currently
+    // not supported by banshee
```

Comment:
Oh, okay, we might want to add this to Banshee, I'll open an issue for it :-)

Reply:
I am not 100% sure if banshee really doesn't support atomics to global memory. But I noticed that atomics to the global memory are definitely not coherent in the case of multi-cluster systems. So I assumed that it is not supported

Reply:
Atomics are supported in Banshee as well. What behavior are you observing? Can you provide me with a binary or test to debug this?

Reply:
Interestingly, I cannot reproduce the issue anymore :smile: 
I had the issue that the barrier on the main memory was treated as if it was private to each cluster i.e. the barrier did not increment to the number of cores globally but only to cores per cluster.

Currently, the atomics are working correctly when running banshee with `--trace`. However, it does not terminate when running without the trace option. Do you have an idea why this might be?

I am running the `printf_fmt` test from this branch:
`banshee --num-clusters 2 --num-cores 9 --no-opt-jit --no-opt-llvm --trace sw/build/snRuntime/test-snRuntime-printf_fmtint > log.dasm`

Reply:
Ok good to hear. The problem might have been fixed in PR https://github.com/pulp-platform/snitch/pull/188. The address map of Banshee was not aligned, which could have lead to the TCDM and global memory being swapped.

That it works with `--trace` sounds a bit like a race condition to me. I can have a closer look.

Reply:
That sounds like the problem I had. I have re-built banshee with the new version at some point. This is would also explain why I can't reproduce the issue anymore